src/opengl.c

```diff
@@ -415,6 +434,225 @@ bool glx_init_blur(session_t *ps) {
+		    "u_resolution.y-u_texsize.y-u_texcoord.y);\n"
+		    "  vec4 u_v4SrcColor = %s(tex_scr, vec2(gl_TexCoord[0].st));\n"
+		    "\n";
+
+		// Fragment shader (round corners)
+		// dst0 shader
+		static const char *FRAG_SHADER_ROUND_CORNERS_0 =
+		    "  float u_fRadiusPx = u_radius;\n"
+		    "  float u_fHalfBorderThickness = 0.0;\n"
+		    "  vec4 u_v4BorderColor = vec4(1.0, 0.0, 0.0, 1.0);\n"
+		    "  vec4 u_v4FillColor = vec4(0.0, 0.0, 0.0, 0.0);\n"
+		    "  vec4 v4FromColor = u_v4BorderColor;	//Always the border "
+		    "color. If no border, this still should be set\n"
+		    "  vec4 v4ToColor = u_v4SrcColor;		//Outside color is the "
+		    "background texture\n"
+		    "\n"
+		    "  vec2 u_v2HalfShapeSizePx = u_texsize/2.0 - "
+		    "vec2(u_fHalfBorderThickness);\n"
+		    "  vec2 v_v2CenteredPos = (gl_FragCoord.xy - u_texsize.xy / 2.0 - "
+		    "coord);\n"
+		    "\n"
+		    "  float fDist = RectSDF(v_v2CenteredPos, u_v2HalfShapeSizePx, "
+		    "u_fRadiusPx - u_fHalfBorderThickness);\n"
+		    "  if (u_fHalfBorderThickness > 0.0) {\n"
+		    "    if (fDist < 0.0) {\n"
+		    "      v4ToColor = u_v4FillColor;\n"
+		    "    }\n"
+		    "    fDist = abs(fDist) - u_fHalfBorderThickness;\n"
+		    "  } else {\n"
+		    "    v4FromColor = u_v4FillColor;\n"
+		    "  }\n"
+		    "  float fBlendAmount = clamp(fDist + 0.5, 0.0, 1.0);\n"
+		    "  vec4 c = mix(v4FromColor, v4ToColor, fBlendAmount);\n"
+		    "\n"
+		    "  // final color\n"
+		    "  gl_FragColor = c;\n"
+		    "\n"
+		    "}\n";
+
+		// Fragment shader (round corners)
+		// dst1 shader
+		static const char *FRAG_SHADER_ROUND_CORNERS_1 =
+		    "  float u_fRadiusPx = u_radius;\n"
+		    "  float u_fHalfBorderThickness = 10.0 /2.0;//0.0;\n"
+		    "  vec4 u_v4BorderColor = vec4(1.0, 0.0, 0.0, 1.0);\n"
+		    "  vec4 u_v4FillColor = vec4(0.0, 1.0, 0.0, 1.0);\n"
+		    "  vec4 v4FromColor = u_v4BorderColor; //Always the border color. If "
+		    "no border, this still should be set\n"
+		    "  vec4 v4ToColor = vec4(0.0, 0.0, 1.0, 1.0); //Outside color\n"
+		    "\n"
+		    "  if (u_is_focused > 0) u_fHalfBorderThickness = u_borderw / 2.0 * "
+		    "4.02;\n"
+		    "  vec2 u_v2HalfShapeSizePx = u_texsize/2.0 - "
+		    "vec2(u_fHalfBorderThickness);\n"
+		    "  vec2 v_v2CenteredPos = (gl_FragCoord.xy - u_texsize.xy / 2.0 - "
+		    "coord);\n"
+		    "\n"
+		    "  float fDist = RectSDF(v_v2CenteredPos, u_v2HalfShapeSizePx, "
+		    "u_fRadiusPx - u_fHalfBorderThickness);\n"
+		    "  if (u_fHalfBorderThickness > 0.0) {\n"
+		    "    if (fDist < 0.0) {\n"
+		    "      v4ToColor = u_v4FillColor;\n"
+		    "    }\n"
+		    "    fDist = abs(fDist) - u_fHalfBorderThickness;\n"
+		    "  } else {\n"
+		    "    v4FromColor = u_v4FillColor;\n"
+		    "  }\n"
+		    "  float fBlendAmount = clamp(fDist + 0.5, 0.0, 1.0);\n"
+		    "  vec4 c = mix(v4FromColor, v4ToColor, fBlendAmount);"
+		    "\n"
+		    "  // final color\n"
+		    "  gl_FragColor = c;\n"
+		    "}\n";
```

Comment:
This renders solid-color windows and looks like being used for debugging.

Reply:
@tryone144 correct.